### PR TITLE
feat(platform): add child handling callbacks

### PR DESCRIPTION
**Pull request by thcu-gp**

### CODE REVIEW

**Comment by thcu-gp** [comment]
ping @jwhui

File: src/qpg7015m/spinel_extension.cpp [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+    {
+        // TODO: Implement your property get handlers here.
+        //
+        // Get handler should retrieve the property value and then encode and write the
+        // value into the NCP buffer. If the "get" operation itself fails, handler should
+        // write a `LAST_STATUS` with the error status into the NCP buffer. `OT_ERROR_NO_BUFS`
+        // should be returned if NCP buffer is full and response cannot be written.
+    case SPINEL_PROP_VENDOR_NEIGHBOR_TABLE_CALLBACK:
+    {
+        bool enable = ((NcpVendorUart *)(ot::Ncp::NcpBase::GetNcpInstance()))->mRegisterCallback;
```
Comment (abtink) [inline src/qpg7015m/spinel_extension.cpp:119]: I think can do this instead:
```suggestion
        bool enable = ((NcpVendorUart *)(this)->mRegisterCallback;
```

**Review by abtink — APPROVED** [review]
Looks good overall.

**Comment by thcu-gp** [comment]
@abtink @jwhui who can merge this?

**Pull request merged**